lib/hashes.md

```diff
@@ -5,40 +5,93 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is like an array and separates a group of values into a list, but unlike an array, hashes identify each element value by an associated key.
+
+
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Inventory information would be useful to be stored as Hashes, such as produce items, or apparel.
+
+
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Similar to an array, a hash separates a group of values into a list, but different from an array, hashes identify each element value by an associated key and finds them using a key, rather than relying on a value's position in a list, like an array does.
+
+
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+You retrieve a hash value by referencing the key you want to retrieve the value from:
+
+1) parts = {"alternators" => 8, "starters" => 16, "mufflers" => 7, "brake rotors" => 22}
+2) puts parts["alternators"]   
+(Result)=> 8
+
+
+
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+You can add information to a has by using the ".merge" method on an existing hash:
+
+1) parts = {"alternators" => 8, "starters" => 16, "mufflers" => 7, "brake rotors" => 22}
+    (Result) => {"alternators"=>8, "starters"=>16, "mufflers"=>7, "brake rotors"=>22}
+
+2) parts.merge("spark plugs" => 38)
+    (Result) => => {"alternators"=>8, "starters"=>16, "mufflers"=>7, "brake rotors"=>22, "spark plugs"=>38}
+
```

Comment:
Nice! So does running `parts.merge("spark plugs" => 38)` permanently alter the `parts` hash?

What is a scenario where you might want to merge two hashes together without altering the original? What are scenario where you might want that change to be reflected in the original hash?

How would you go about adding information to a hash such that the hash is permanently altered?